internal/client/client.go

```diff
@@ -108,6 +108,21 @@ func (c *CSAPI) InviteRoom(t *testing.T, roomID string, userID string) {
 	c.MustDo(t, "POST", []string{"_matrix", "client", "r0", "rooms", roomID, "invite"}, body)
 }
 
+// SearchUserDirectory makes a request to search the user directory with a given query.
+// We don't include an explicit limit on the number of results returned, relying on
+// the spec's default of 10.
+func (c *CSAPI) SearchUserDirectory(t *testing.T, query string) *http.Response {
```

Comment:
Please don't put infrequently used endpoints in `CSAPI`. See https://github.com/matrix-org/complement/pull/146#discussion_r663866719 - it hurts readability/usability when everyone adds their specific endpoint to `CSAPI` when it isn't used in many other tests, and it also encourages bad API shapes (e.g right now this function accepts `query string` but explicitly doesn't include a `limit`, but later tests may want to adjust the limit, meaning an API change).

Given this function is literally a proxy function, just call `MustDoFunc` directly please.